flang/unittests/Lower/OpenMPLoweringTest.cpp

```diff
@@ -71,4 +71,38 @@ TEST_F(OpenMPLoweringTest, TaskYield) {
   EXPECT_EQ(succeeded(taskYieldOp.verify()), true);
 }
 
+TEST_F(OpenMPLoweringTest, EmptyParallel) {
+  // Construct a dummy parse tree node for `!OMP parallel`.
+  struct Fortran::parser::OmpSimpleStandaloneDirective parallelDirective(
+      llvm::omp::Directive::OMPD_parallel);
+
+  // Check and lower the `!OMP parallel` node to `ParallelOp` operation of
+  // OpenMPDialect.
+  EXPECT_EQ(parallelDirective.v, llvm::omp::Directive::OMPD_parallel);
```

Comment:
The unit test looks a bit weird to me, it looks like to me it is testing a copy paste of the code from `genOMP` rather than calling `genOMP` directly or indirectly and checking the side effects have the intended properties. Its hard to ensure that the code inside the tests is and will remain the same as the one in `genOMP`. It looks a good unit test for OpenMP dialect however (testing ParallelOp  can be created as intended, regardless of Flan/Fortran/Fir context).

Reply:
> I am a bit surprised by the way the operation has to be created manipulating the opState here.
Why is it no possible to do the following ?
Thanks, I was facing some obscure issue with this sort of creation(May be due to unclean environment). Now it's working great!

Reply:
> The unit test looks a bit weird to me, it looks like to me it is testing a copy paste of the code from genOMP rather than calling genOMP directly or indirectly and checking the side effects have the intended properties. Its hard to ensure that the code inside the tests is and will remain the same as the one in genOMP. It looks a good unit test for OpenMP dialect however (testing ParallelOp can be created as intended, regardless of Flan/Fortran/Fir context).
This unit-test is meant to be upstreamed. And the regression test will remain in fir-dev. I'm not sure whether all the pieces of bridge are there in upstream FLANG to call functionally call genOMP.
Besides this unit-test is meant for operation creation and validation. Once the operation is created correctly it can be lowered.